### PR TITLE
Build and push community temp index image to Quay repository

### PR DESCRIPTION
**Pull request by Allda**
The community hosted pipeline is extended with a build index step that takes an input bundle and adds it into all supported indexes. The new index is pushed into Quay repository where it can be use by Prow jobs.

JIRA: ISV-3505

### CODE REVIEW

File: operator-pipeline-images/operatorcert/entrypoints/index.py [hunk 1]
```diff
@@ -149,11 +170,42 @@ def output_index_image_paths(
...
+    auth_file: Optional[str] = None,
+) -> None:
+    """
+    Copy IIB index images to a destination given by parameter
+
+    Args:
+        iib_response (Any): IIB build response object
+        destination (str): Registry and repository destination for the index images
+        tag_suffix (str): Tag suffix to append to the index image
+        auth_file (Optional[str], optional): Auth file for destination registyr.
```
Comment (mporrato) [inline operator-pipeline-images/operatorcert/entrypoints/index.py:191]:
```suggestion
        auth_file (Optional[str], optional): Auth file for destination registry.
```

**Review by mporrato — APPROVED** [review]

**Review by haripate — APPROVED** [review]
LGTM

File: operator-pipeline-images/operatorcert/entrypoints/index.py (hunk 1 above)
Comment (Allda) [reply operator-pipeline-images/operatorcert/entrypoints/index.py]: Fixed

File: ansible/roles/operator-pipeline/templates/openshift/tasks/add-bundle-to-index.yml [hunk 2]
```diff
@@ -6,22 +6,38 @@ metadata:
...
       description: >-
         The key within the Kubernetes Secret that contains the kerberos keytab for submitting IIB builds.
+
     - name: environment
       description: |
         Which environment the pipeline is running in. Can be one of [dev, qa, stage, prod]
+
+    - name: index-image-destination
+      default: ""
+      description: "The destination registry and repository when index images will be pushed after IIB build is finished"
```
[the remark this answers is not in the document]
Comment (Allda) [reply ansible/roles/operator-pipeline/templates/openshift/tasks/add-bundle-to-index.yml]: Fixed

**Pull request merged**